pkg/storaged/filesystem/mismounting.jsx

```diff
@@ -178,9 +178,9 @@ export const MismountAlert = ({ warning, fstab_config, forced_options, backing_b
         else if (type == "mount-on-boot")
             return do_mount();
         else if (type == "no-mount-on-boot")
-            return do_unmount();
+            return do_unmount(old_dir);
         else if (type == "mounted-no-config")
-            return do_unmount();
+            return do_unmount(old_dir || other);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19697-20231205-151550-901259c1-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)